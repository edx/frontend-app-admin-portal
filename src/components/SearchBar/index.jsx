import React from 'react';
import PropTypes from 'prop-types';
import { SearchField } from '@edx/paragon';

class SearchBar extends React.Component {
  handleQuerySubmit(query) {
    this.props.onSearch(query);
  }

  render() {
    return (
      <SearchField
        onSubmit={query => this.props.onSearch(query)}
        {...this.props}
      />
    );
  }
}

SearchBar.propTypes = {
  onSearch: PropTypes.func.isRequired,
};

export default SearchBar;
