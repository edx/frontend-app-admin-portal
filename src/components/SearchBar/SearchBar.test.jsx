import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import SearchBar from './index';

describe('<SearchBar />', () => {
  let wrapper;

  it('renders correctly', () => {
    const tree = renderer
      .create((
        <SearchBar
          onSearch={() => {}}
        />
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onSearch callback handler', () => {
    const mockOnSearchCallback = jest.fn();
    wrapper = mount((
      <SearchBar
        onSearch={mockOnSearchCallback}
      />
    ));
    wrapper.find('input[type=\'search\']').simulate('change', { target: { value: 'foobar' } });
    wrapper.find('.input-group-append').find('button.search-btn').simulate('click');
    expect(mockOnSearchCallback).toHaveBeenCalledTimes(1);
  });
});
