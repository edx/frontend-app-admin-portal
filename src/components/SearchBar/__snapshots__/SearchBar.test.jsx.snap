// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> renders correctly 1`] = `
<div
  className="border search-field"
  onBlur={[Function]}
  onFocus={[Function]}
>
  <div
    className="form-group form-inline"
  >
    <label
      className=""
      htmlFor="asInput3"
      id="label-asInput3"
    >
      Search:
    </label>
    <div
      className="input-group"
    >
      <div
        className="input-group-prepend"
      />
      <input
        aria-describedby="error-asInput3"
        aria-invalid={false}
        autoComplete="off"
        className="form-control is-invalid-nodanger input no-clear-btn"
        disabled={false}
        id="asInput3"
        name="search"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder=""
        readOnly={false}
        required={false}
        themes={Array []}
        type="search"
        value=""
      />
      <div
        className="input-group-append"
      >
        <button
          className="btn search-btn"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <span
            aria-hidden={true}
            className="fa fa-search"
            id="Icon1"
          />
          <span
            className="sr-only"
          >
            Submit search
          </span>
        </button>
      </div>
    </div>
    <div
      aria-live="polite"
      className="invalid-feedback invalid-feedback-nodanger"
      id="error-asInput3"
    >
      <span />
    </div>
  </div>
</div>
`;
